packages/perseus-editor/src/widgets/python-program-editor.tsx

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable no-useless-escape */
+/**
+ * This editor is for embedding Khan Academy Python programs.
+ */
+
+import {
+    components,
+    Changeable,
+    EditorJsonify,
+} from "@khanacademy/perseus";
+import PropTypes from "prop-types";
+import * as React from "react";
+import _ from "underscore";
+
+import BlurInput from "../components/blur-input";
+
+const {NumberInput} = components;
+
+const DEFAULT_HEIGHT = 400;
+
+type Props = any;
+
+/**
+ * This is the main editor for this widget, to specify all the options.
+ */
+class PythonProgramEditor extends React.Component<Props> {
+    static propTypes = {
+        ...Changeable.propTypes,
+        programID: PropTypes.number,
+        height: PropTypes.number,
+    };
+
+    static widgetName = "python-program" as const;
+
+    static defaultProps: any = {
+        programID: null,
+        height: DEFAULT_HEIGHT,
+    };
+
+    change: (...args: ReadonlyArray<unknown>) => any = (...args) => {
+        // @ts-expect-error - TS2345 - Argument of type 'readonly unknown[]' is not assignable to parameter of type 'any[]'.
+        return Changeable.change.apply(this, args);
+    };
+
+    _handleHeightChange: (arg1: string) => void = (height) => {
+        this.change({height});
+    };
+
+    _handleProgramIDChange: (arg1: string) => void = (programID) => {
+        this.change({programID});
+    };
+
+    serialize: () => any = () => {
+        return EditorJsonify.serialize.call(this);
+    };
```

Comment:
We don't have a pattern for this encoded very clearly yet, but I'd love for this function to:

1) clearly define it's return type. It should be the type that is defined in `perseus-types.ts` as it's the type that is serialized as widget options to Datastore. 

2) Avoid using `EditorJsonify` and just manually, and clearly, build whatever the return value should be. We (I) have found that this dynamism just complicates things and obscures which properties we're serializing, which also contributes to properties being serialized that don't really need to be. 

```suggestion
    serialize(): PerseusPythonProgramWidgetOptions {
        return {
            programID: this.props.programID,
            height: this.props.height,
        };
    } 
```